libs/qCC_db/ccGBLSensor.cpp

```diff
@@ -1051,8 +1051,8 @@ bool ccGBLSensor::fromFile_MeOnly(QFile& in, short dataVersion, int flags)
 	ccSerializationHelper::CoordsFromDataStream(inStream, flags, &m_deltaTheta, 1);
 	if (dataVersion < 38)
 	{
-		ccSerializationHelper::ScalarsFromDataStream(inStream, flags, &m_sensorRange, 1);
-		ccSerializationHelper::ScalarsFromDataStream(inStream, flags, &m_uncertainty, 1);
+		ccSerializationHelper::ScalarsFromDataStream(inStream, flags, static_cast<ScalarType*>(&m_sensorRange), 1);
```

Comment:
Actually there was an error here. m_sensorRange and m_uncertainty are declared as PointCoordinateType. Therefore one should use CoordsFromDataStream to read them.

Reply:
This would somehow conflict with what seems to be some logic here. The `dataVersion` variable seems to delimit a specific range of data format versions, where the data is to be read differently.

Making the suggested changes, I could basically remove the `if` statement surrounding this together with `else` case.

Reply:
Hey @dgirardeau, any movements on this?

Reply:
Sorry I was in vacations. To be on the safe side, we should then 'cast' the m_sensorRange and m_uncertainty variables to ScalarType ones before saving them. And same thing, read them with the ScalarsFromDataStream and temporary ScalarType variables.

Reply:
Would the following be correct?
```cpp
ScalarType sensorRange{}, uncertainty{};
ccSerializationHelper::ScalarsFromDataStream(inStream, flags, &sensorRange, 1);
ccSerializationHelper::ScalarsFromDataStream(inStream, flags, &uncertainty, 1);
m_sensorRange = static_cast<PointCoordinateType>(sensorRange);
m_uncertainty = static_cast<PointCoordinateType>(uncertainty);
```

As I understand, the temporary `ScalarType` values are there to secure us from reading too much from the buffer (into wrong memory), but I might not be getting this part of your request correctly:
> we should then 'cast' the m_sensorRange and m_uncertainty variables to ScalarType ones before saving them

Reply:
Yes I think it's good like that.